Sources/EngineKit/HashableExtensions.swift

```diff
@@ -522,14 +782,32 @@ extension TransactionType: Hashable {
 				lhsMetadataOfNewlyCreatedEntities == rhsMetadataOfNewlyCreatedEntities &&
 				lhsDataOfNewlyMintedNonFungibles == rhsDataOfNewlyMintedNonFungibles &&
 				lhsAddressesOfNewlyCreatedEntities == rhsAddressesOfNewlyCreatedEntities
+		case (.generalTransaction, _):
+			return false
+
 		case let (
 			.accountDepositSettings(lhsResourcePreferenceChanges, lhsDefaultDepositRuleChanges, lhsAuthorizedDepositorsChanges),
 			.accountDepositSettings(rhsResourcePreferenceChanges, rhsDefaultDepositRuleChanges, rhsAuthorizedDepositorsChanges)
 		):
 			return lhsResourcePreferenceChanges == rhsResourcePreferenceChanges &&
 				lhsDefaultDepositRuleChanges == rhsDefaultDepositRuleChanges &&
 				lhsAuthorizedDepositorsChanges == rhsAuthorizedDepositorsChanges
-		default:
+		case (.accountDepositSettings, _):
+			return false
+
+		case let (.claimStakeTransaction(lhsClaims), .claimStakeTransaction(rhsClaims)):
+			return lhsClaims == rhsClaims
+		case (.claimStakeTransaction, _):
+			return false
+
+		case let (.stakeTransaction(lhsStakes), .stakeTransaction(rhsStakes)):
+			return lhsStakes == rhsStakes
+		case (.stakeTransaction, _):
+			return false
+
+		case let (.unstakeTransaction(lhsUnstake), .unstakeTransaction(unstakeTransaction)):
+			return lhsUnstake == unstakeTransaction
```

Comment:
the rhs value is not named `rhsUnstake`, is that on purpose?

Reply:
accidental, will fix tomorrow, in another PR, copy paste error.